### PR TITLE
feat: adding app_id support in braze

### PR DESCRIPTION
**Pull request by shrouti1507**
## Description of the change

> Adding support for sending app_id in braze events.
https://www.braze.com/docs/api/identifier_types/?tab=app%20ids

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/braze/util.js [hunk 1]
```diff
@@ -445,10 +444,38 @@ const processBatch = (transformedEvents) => {
...
+const addAppId = (payload, message) => {
+  const integrationsObj = getIntegrationsObj(message, 'BRAZE');
+  if (integrationsObj?.appId) {
+    const { appId: appIdValue } = integrationsObj;
+    return {
+      ...payload,
+      app_id: String(appIdValue),
+    };
+  }
+  return { ...payload };
```
Comment (koladilip) [inline src/v0/destinations/braze/util.js:471]: why not simply `return payload`?

File: src/v0/destinations/braze/transform.js [hunk 2]
```diff
@@ -401,6 +402,7 @@ function processTrackEvent(messageType, message, destination, mappingJson, proce
   payload.properties = properties;
 
   payload = setExternalIdOrAliasObject(payload, message);
+  payload = addAppId(payload, message);
   if (payload) {
```
Comment (koladilip) [inline src/v0/destinations/braze/transform.js:406]: is it necessary?

**Review by koladilip — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

File: src/v0/destinations/braze/transform.js (hunk 2 above)
Comment (shrouti1507) [reply src/v0/destinations/braze/transform.js:406]: yes, this is what support they asked for

Comment (koladilip) [reply src/v0/destinations/braze/transform.js:406]: I am asking about checking for payload being not defined. as in line 402 we are setting properties on payload

**Comment by shrouti1507** [comment]
closing this PR. We are raising a new one against hotfix branch

**Pull request closed without merging**